Add mimxrt118x src support

zejiang0jason opened the pull request:
Make the hwinfo.mcux_src_rev2 zephyr native driver.
Add RT118x SRC driver support.

zejiang0jason committed:
drivers: hwinfo: mcux_src_rev2: Update to support MIMXRT118X

Add RT118X support

Signed-off-by: Jason Yu <zejiang.yu@nxp.com>

zejiang0jason committed:
drivers: hwinfo: mcux_src_rev2: Change it to native driver

The reset source is SOC specific, MCUX SDK SRC driver is
also SOC specific, it doesn't mask the SOC differences.
So the driver can use SOC macros directly, and handle SOC
differences in the hwinfo driver, not necessary to wrap
MCUX SDK driver.

Signed-off-by: Jason Yu <zejiang.yu@nxp.com>

zejiang0jason commented on soc/nxp/imxrt/imxrt118x/Kconfig:
```diff
@@ -21,6 +21,7 @@ config SOC_SERIES_IMXRT118X
 	select CPU_HAS_FPU_DOUBLE_PRECISION if SOC_MIMXRT1189_CM7
 	select HAS_SWO
 	select SOC_EARLY_INIT_HOOK
+	select HAS_MCUX_SRC_V2
```
This is used in other SOCs like: 
https://github.com/zephyrproject-rtos/zephyr/blob/0bc5f7d4dbd026f2e938bb4104de7ab079e85dff/soc/nxp/imxrt/imxrt11xx/Kconfig#L17

Do you mean remove them or replace them to another name.

zejiang0jason replied:
right, there is not dts binding file.

decsny replied:
@zejiang0jason can you take this opportunity to make that file and convert this to remove another one of these type of config ?

zejiang0jason committed:
drivers: hwinfo: mcux_src_rev2: Change to use dts as dependency

Originally the driver is selected if `HAS_MCUX_SRC_V2` is
selected in SOC level kConfig.
Change to use dts to mark the driver is avaiable for some SOC.

Signed-off-by: Jason Yu <zejiang.yu@nxp.com>
